### PR TITLE
fix:fix response format bug

### PR DESCRIPTION
**Pull request by guaguaguaxia**
fix this issue:https://github.com/openai/openai-python/issues/243

### CODE REVIEW

File: openai/api_requestor.py [hunk 1]
```diff
@@ -666,7 +667,10 @@ def _interpret_response_line(
                 headers=rheaders,
             )
         try:
-            data = json.loads(rbody)
+            if self.response_format == ResponseFormat.JSON:
+                data = json.loads(rbody)
+            else:
+                data = rbody
```
Comment (MattFisher) [inline openai/api_requestor.py:673]: I checked and the Content-Type header on the response is always set correctly (at least for the audio transcription responses), so we can deduce whether to json-decode without any other code changes.
For the `json` and `verbose_json` response formats, the Content-Type is 'application/json', while for `text, `srt`, and `vtt`, the Content-Type is `'text/plain; charset=utf-8'`.

```suggestion
            if rheaders.get('Content-Type') == 'application/json':
                data = json.loads(rbody)
            else:
                data = rbody
```

**Comment by guaguaguaxia** [comment]
@MattFisher please review in https://github.com/openai/openai-python/pull/282, i don't know how i close this PR 😭

**Comment by guaguaguaxia** [comment]
@mpokrass

**Comment by guaguaguaxia** [comment]
@logankilpatrick  @mpokrass

**Pull request closed without merging**